### PR TITLE
Finish subquery history

### PR DESCRIPTION
**Pull request by valentunn**

### CODE REVIEW

File: common/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <resources>
+    <string name="staking_rewards_event">Event</string>
+    <string name="common_copy_event_id">Copy Event Id</string>
+    <string name="staking_slash_details">Slash details</string>
```
Comment (antonkhvorov) [inline common/src/main/res/values/strings.xml:5]: We can just use "Details" title for all details screen (Transfer/Reward & Slash /Extrinsic)

File: feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/di/WalletFeatureModule.kt [hunk 2]
```diff
@@ -55,8 +57,14 @@ class WalletFeatureModule {
 
     @Provides
     @FeatureScope
-    fun provideSubscanApi(networkApiCreator: NetworkApiCreator): WalletNetworkApi {
-        return networkApiCreator.create(WalletNetworkApi::class.java)
+    fun provideSubscanApi(networkApiCreator: NetworkApiCreator): SubQueryOperationsApi {
```
Comment (antonkhvorov) [inline feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/di/WalletFeatureModule.kt:60]: Do we need it still?

Comment (antonkhvorov) [reply feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/di/WalletFeatureModule.kt:60]: Oh wait the name of the function is wrong, should be "provideSubQueryApi"

File: feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/transaction/history/mixin/TransactionStateMachine.kt [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+import jp.co.soramitsu.common.data.model.CursorPage
+import jp.co.soramitsu.feature_wallet_api.domain.interfaces.TransactionFilter
+import jp.co.soramitsu.feature_wallet_api.domain.interfaces.allFiltersIncluded
+import jp.co.soramitsu.feature_wallet_api.domain.model.Operation
+
+object TransactionStateMachine {
+
+    const val PAGE_SIZE = 100
+    private const val SCROLL_OFFSET = PAGE_SIZE / 2
```
Comment (antonkhvorov) [inline feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/transaction/history/mixin/TransactionStateMachine.kt:11]: why not keep it 100?

**Review by antonkhvorov — APPROVED** [review]

File: feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/transaction/history/mixin/TransactionStateMachine.kt (hunk 3 above)
Comment (valentunn) [reply feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/transaction/history/mixin/TransactionStateMachine.kt:11]: if pagesize == scrollOffset, then the second page will be triggered to load immediately

Comment (antonkhvorov) [reply feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/transaction/history/mixin/TransactionStateMachine.kt:11]: Got it, looked in the code once again, yeah, 50 is great to have. Thank you

**Pull request merged**